streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedTableImpl.java

```diff
@@ -109,16 +114,42 @@ private void buildAggregate(final ProcessorSupplier<K, Change<V>> aggregateSuppl
         final String sinkName = builder.newProcessorName(KStreamImpl.SINK_NAME);
         final String sourceName = builder.newProcessorName(KStreamImpl.SOURCE_NAME);
         final String funcName = builder.newProcessorName(functionName);
+        final String topic = materialized.storeName() + KStreamImpl.REPARTITION_TOPIC_SUFFIX;
+
+        RepartitionNode.RepartitionNodeBuilder<K, V> repartitionNodeBuilder = RepartitionNode.repartitionNodeBuilder();
 
         buildAggregate(aggregateSupplier,
-                       materialized.storeName() + KStreamImpl.REPARTITION_TOPIC_SUFFIX,
+                       topic,
                        funcName,
-                       sourceName, sinkName);
+                       sourceName,
+                       sinkName
+        );
+
         builder.internalTopologyBuilder.addStateStore(new KeyValueStoreMaterializer<>(materialized)
-                                                              .materialize(), funcName);
+                                                          .materialize(), funcName);
+
+        ProcessorParameters processorParameters = new ProcessorParameters<>(aggregateSupplier, funcName);
+
+        repartitionNodeBuilder.withRepartitionTopic(topic)
```

Comment:
I only have slight preference on this, since it has nothing to do with correctness but just LOC v.s. programming safety :) To me it is OK allowing the GraphNode's fields to be mutable just like `Materialized` etc, i.e. we allow `withXXX` functions to override them while the constructor only accepts a very small set of parameters shared in the abstract `GraphNode`. But I'll leave it to you to decide which do you prefer, so feel free to ignore this comment.

Reply:
I'll refactor the `Builder` classes out and follow the pattern similar to `Materialized` in the 3rd PR